src/parse/init.go

```diff
@@ -55,15 +59,49 @@ func newAspParser(state *core.BuildState) *asp.Parser {
 	return p
 }
 
-func (p *aspParser) Preload(filename string) {
-	p.parser.MustLoadBuiltins(filename, nil)
+// NewParser creates a new parser for the state
+func (p *aspParser) NewParser(state *core.BuildState) {
+	// TODO(jpoole): remove this once we refactor core so it can depend on this package and call this itself
+	state.Parser = nil
+	InitParser(state)
 }
 
-func (p *aspParser) ParseFile(state *core.BuildState, pkg *core.Package, filename string) error {
+func (p *aspParser) WaitForInit() {
+	<-p.init
+}
+
+func (p *aspParser) preloadSubincludes(state *core.BuildState) {
+	includes := state.Config.Parse.PreloadSubincludes
+	if state.RepoConfig != nil {
+		// TODO(jpoole): is this the right thing to do?
+		includes = append(includes, state.RepoConfig.Parse.PreloadSubincludes...)
+	}
```

Comment:
Just trying to get my head around this - can we have both a Config and a RepoConfig on a state? Or does having a RepoConfig imply that the state belongs to a subrepo and so there's no Config?